[R-package] MSVC warning: "MSB8029: The Intermediate directory or Output directory cannot reside under the Temporary directory"

See https://github.com/microsoft/LightGBM/pull/3076#issuecomment-629477537 for background. On R package builds with MSVC and Visual Studio 2019, we see the following warning in the build logs:

> C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(475,5): warning MSB8029: The Intermediate directory or Output directory cannot reside under the Temporary directory as it could lead to issues with incremental build. 

See, for example, [the most recent `master` build](https://ci.appveyor.com/project/guolinke/lightgbm/builds/32890583/job/n9e9f9x0j3hmjtdb)

![image](https://user-images.githubusercontent.com/7608904/82097547-bacf6100-96c8-11ea-8c37-4f3981fb583a.png)

I found [at least one issue](https://github.com/grpc/grpc/pull/4524) where someone claimed to be able to suppress this by adding the following in their `.vcxproj` file:

```xml
<IgnoreWarnIntDirInTempDetected>true</IgnoreWarnIntDirInTempDetected>
```

We don't use one of those configs for the R package, so that won't work.

I haven't found anything about this in official Visual Studio documentation yet.

Marking this `r-package` because it seems that we don't get the error in Visual Studio + CMake builds on the Python side (see [the logs from the most recent run of the `windows sdist` task](https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=6150&view=logs&j=67ae2ff8-4b33-578b-0813-9c0440091667&t=130bfb73-8505-5e4f-f030-9ec1bace59ea)).

## Why this issue matters

I don't know if this will cause any problems for our users. I suspect that it won't. But every warning that shows up in users' logs is something they might think is a reason for concern, and might spend time searching for. I'd like to suppress this if possible to save LightGBM R users from wasting time searching for the meaning of and solution to this warning.

--- a/.ci/test_r_package_windows.ps1
+++ b/.ci/test_r_package_windows.ps1
@@ -121,6 +121,7 @@ if ($env:COMPILER -ne "MSVC") {
       Check-Output $False
   }
 } else {
+  $env:TMPDIR = $env:USERPROFILE  # to avoid warnings about incremental builds inside a temp directory
   $INSTALL_LOG_FILE_NAME = "$env:BUILD_SOURCESDIRECTORY\00install_out.txt"
   Rscript build_r.R *> $INSTALL_LOG_FILE_NAME ; $install_succeeded = $?
   Write-Output "----- build and install logs -----"

